bioconda_utils/lint_functions.py

```diff
@@ -295,18 +295,18 @@ def pin(recipe, meta, df):
         not_pinned = set()
         pinned = set()
         for line in open(os.path.join(recipe, 'meta.yaml')):
+            dedented_line = line.lstrip(' ')
             if line.startswith("requirements:"):
                 in_requirements = True
-            elif in_requirements and line.strip().startswith("run:"):
+            elif in_requirements and dedented_line.startswith("run:"):
                 section = "run"
-            elif in_requirements and line.strip().startswith("build:"):
+            elif in_requirements and dedented_line.startswith("build:"):
                 section = "build"
-            elif not line.startswith(" ") and not line.startswith("#"):
+            elif dedented_line and not line.startswith(' ') and not line.startswith("#"):
                 in_requirements = False
                 section = None
-            line = line.strip()
-            if in_requirements and line.startswith('- {}'.format(dep_name)):
-                if pin_pattern.search(line) is None:
+            if in_requirements and dedented_line.startswith('- {}'.format(dep_name)):
+                if pin_pattern.search(line.strip()) is None:
```

Comment:
`dedented_line` instead of `line.strip()` should suffice as we're only searching a substring with the regex anyway